common/src/main/java/org/apache/uniffle/common/util/RssUtils.java

```diff
@@ -218,6 +220,13 @@ private static List<ShuffleDataSegment> transIndexDataToSegments(byte[] indexDat
 
         bufferSegments.add(new BufferSegment(blockId, bufferOffset, length, uncompressLength, crc, taskAttemptId));
         bufferOffset += length;
+        totalLength += length;
+
+        // If ShuffleServer is flushing the file at this time, the length in the index file record may be greater
```

Comment:
I think this problem only occur that the map tasks are all finished and the data stored in memory is flushed to localfile/HDFS, right? And in this time, the spark client read the redundant index data. Right? 

Analyzed from this perspective, if u drop these redundant data, does it will cause the data missing problem due to in hdfs client buffer insteading of memory/HDFS. I think it wont. The flushing data only will be flushed to HDFS and then removed from memory. And the method of `dataWriter.close()` in `HdfsShuffleWriteHandler` will ensure the data flushed to HDFS.

So this change is OK and wont cause data lose. But I have a question that why not calling the `dataWriter.flush` and `indexWriter.flush` when writing one block to solve this problem. Does this will make performance regession?

Reply:
It seems unreasonable to do a flush every time a block is written, so that the buffer of the hdfs client will not work, and it will make performance regession.